### ISSUE TITLE
License warnings don't use ignore list

### ISSUE THREAD

**Issue by Pradoxzon** [advanced-security/policy-as-code#46]
**Describe the bug**
I'm not sure if this is a true bug in the functionality or if this design was intended.  
Currently, `checkDependencyLicensing()` in `checks.py` [on line 227](https://github.com/advanced-security/policy-as-code/blob/main/ghascompliance/checks.py#L227) does not use the ignore lists for warnings. The list of warnings is created and handled before the ignore lists are created.

**To Reproduce**
1. Create a licensing policy that contains a list of licenses for `warnings`, `conditions`, and `ignores`
2. Ensure the ignores list shares at least 1 license with the warnings and conditions lists
3. Run the policy on a repo that contains dependencies with the license(s) in the ignores list
4. Observe that warnings are still generated for the license(s) in the ignores list

Example policy:
```yml
name: License Compliance Testing Policy

general:
  level: none

licensing:
  warnings:
    ids:
      - MIT
      - MIT-0
      - BSD-1-Clause
      - BSD-2-Clause
      - BSD-3-Clause
      - Other
      - NA

  conditions:
    ids:
      - MIT
      - MIT-0
      - BSD-1-Clause
      - BSD-2-Clause
      - BSD-3-Clause
      - Other
      - NA

  ignores:
    ids:
      - MIT
```

**Expected behavior**
As a user, I would generally expect an ignore list to apply to apply to both warnings and violations. I don't need to see a warning for a dependency/license that I already know I want to ignore.

**Build**
GitHub Actions, Policy as Code version 2.3.7, using ubuntu-latest

**Comment by Pradoxzon** [comment]
I realize this is probably irrelevant with the changes coming in v3, the functionality of having the ignore list apply to warnings as well as violations could still be an enhancement.

**Comment by GeekMasher** [comment]
@Pradoxzon I found the issue around this, there was a typo in the code and why it wasn't picking up any ignored items.

https://github.com/advanced-security/policy-as-code/issues/55#issuecomment-1658716687

Please update to v2.4.1 and let me know if its resolved the issue.

**Comment by GeekMasher** [comment]
@Pradoxzon Can you confirm if this patch fixed your bug so I can close this issue?

**Comment by Pradoxzon** [comment]
@GeekMasher sorry for the delayed response, it looks like the bug in the logic is still present in v2.4.1.

In [`checkDependencyLicensing()` in checks.py](https://github.com/advanced-security/policy-as-code/blob/d4e9cbd5ea0eb8d05ac22db0eae3f2a78b5d29fb/ghascompliance/checks.py#L287), the warning lists (`warnings_ids` and `warnings_names`) are built and added to `licensing_warnings` and logged before the ignore lists are built.  
This can lead to warnings being output for licenses that a user would have expected to be ignored.

My recommendation would be to move the creation of the ignore lists to earlier in the function and apply the same filtering logic to the warning lists that is used for the violation lists later in the funciton.

**Comment by Pradoxzon** [comment]
Fixed with #58.

**Comment by GeekMasher** [comment]
#60 will be released shortly
